src/olympia/reviewers/models.py

```diff
@@ -798,6 +801,52 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @classmethod
+    def get_base_addons(cls):
+        """Return base queryset of add-ons we consider when we look at growth
+        thresholds.
+
+        This is a classmethod, it is not specific to a particular tier."""
+        return Addon.unfiltered.exclude(status=amo.STATUS_DISABLED).filter(
+            type=amo.ADDON_EXTENSION
+        )
+
+    def get_tier_boundaries(self):
+        """Return boundaries used to filter add-ons for that tier instance."""
+        return {
+            'average_daily_users__gte': self.lower_adu_threshold,
+            'average_daily_users__lt': self.upper_adu_threshold,
+        }
+
+    @cached_property
+    def average_growth(self):
+        return (
+            self.get_base_addons()
```

Comment:
Does this result in the same number of queries in the end? It's a bit hard to tell given the refactor and changes to the logic.

Reply:
Yes. That's why I added a `get_growth_threshold_q_object()` method and made `get_base_addons()` separate. We're looping over each `UsageTier`, computing the average for each (so one query per tier - we don't have a lot of them fortunately) and building the final queryset, then once we're done looping we execute the query to find all add-ons over threshold in one single query regardless of which tier they belong to.

Reply:
3e0a058768e81ea1ad6f33b691ccbb6b0d8817e4 adds a test around the number of queries for the task itself